### DISCUSSION TITLE
Doesn't work when using single-word package names?

### DISCUSSION THREAD

**Discussion by danawoodman** [vektra/mockery#758] [Q&A]
I'm probably doing something wrong, but with my package `lprd` (`module lprd`), I'm getting an error that seems to imply mockery is looking in the std lib for my package and not the current workspace. What am I doing wrong?

```yaml
with-expecter: True
packages:
  lprd: 
    all: True
```
I get the following error when running `mockery`:

```shell
12 Feb 24 16:50 PST ERR encountered error when loading package error="-: package lprd is not in std (/usr/local/go/src/lprd)" dry-run=false version=v2.40.3
```

Using Go v1.22.0 and

**Comment by danawoodman** [comment]
And a follow up question, how can I make mockery generate code with a package name like `package foo_test` so that my test files can import the mocks? I don't want a separate test/mock directory

**Reply by LandonTClipp** [reply]
You can control every aspect of what your file and packages are named. Lots of parameters here: https://vektra.github.io/mockery/v2.41/configuration/#parameter-descriptions

**Comment by LandonTClipp** [comment]
I don't know for sure but I think Go assumes that non-qualified names are part of the stdlib. This would not be a mockery-specific behavior. What happens if you put `github.com/danawoodman/lprd` as your module name?

**Reply by LandonTClipp** [reply]
This turns out to indeed be the case. Dotless module names are considered to be part of the stdlib: https://github.com/golang/go/blob/go1.21.0/src/cmd/go/internal/modload/import.go#L38-L42

The comments say a replace directive might be used to get around this.

**Reply by danawoodman** [reply]
hmm, that's bizarre considering i have no issues actually using that outside of mockery (building, testing, etc). I'm assuming the only way to get around this would be to do something like you mention with github as the package qualified package URL?

**Reply by LandonTClipp** [reply]
> I'm assuming the only way to get around this would be to do something like you mention with github as the package qualified package URL?

I guess so, but the comments in the Go source code suggest you can do something with a `replace` directive. Without knowing how your project is physically laid out, or how you're calling mockery, I can't really do much more than speculate. I would be more effective if I had a reproducer to work with.

**Reply by danawoodman** [reply]
i would imagine if you made a blank project named `lprd` or whatever and had an interface you could repro? if i have time i can try and put something together, tho no idea when that will be lol
